internal/os/os_test.go

```diff
@@ -0,0 +1,82 @@
+package os
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+)
+
+func TestCli_OperatingSystem(t *testing.T) {
+	subCommand := "operating-systems"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get", "-h"})
+				root.SetOutput(out)
+				root.Print()
+				if err := root.Execute(); err != nil {
+					t.Error("expected an error")
+				}
+				fmt.Print(out.String())
+				if !strings.Contains(out.String(), "Retrieves a list of operating systems available to the current user") {
+					t.Error("expected output should include list all known operating systems")
+				}
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				t.Helper()
+				root := c.Root()
+				out := &bytes.Buffer{}
+				root.SetArgs([]string{subCommand, "get", "--token", "7UuyGp9itGhqr6LejYYFYxo8ohzpNbBn"})
+				root.SetOutput(out)
+				err := root.Execute()
+				if err != nil {
+					t.Skip()
+					// t.Error("expected an error")
```

Comment:
I think your old logic to fail the test on error is preferable to skipping the test.

```suggestion
					t.Error("an unexpected error occured", err)
```